src/gui/mzroll/ligandwidget.cpp

```diff
@@ -403,25 +404,38 @@ void LigandWidget::showTable() {
                 QTreeWidgetItem *item = new QTreeWidgetItem(parent, PathwayType);
                 item->setText(0,QString(compound->category[i].c_str()));
                 //parent->setData(0,Qt::UserRole,QVariant::fromValue(QString(pathway_id.c_str())));
-
             }*/
 
     }
-    treeWidget->setSortingEnabled(true);
+  
+    
+    Hash.clear();
+    QTreeWidgetItemIterator itr(treeWidget);
+        while(*itr)
+        {     
+        QTreeWidgetItem* item =(*itr);
+        QVariant v = item->data(0,Qt::UserRole);
+        Compound*  c =  v.value<Compound*>();                                                                                   
+        Hash.insert(c , item);
+        ++itr;
+        }
+treeWidget->setSortingEnabled(true);
 }
 
 void LigandWidget::markAsDone(Compound* compound) {
-
-    if (compound != NULL) {
-        QTreeWidgetItem* item = getItem(compound);
-
-        if (item != NULL) {
-            for (int col = 0; col < treeWidget->columnCount(); col++) {
-                item->setBackgroundColor(col, QColor(61, 204, 85, 100));
-            }
+    if(compound != NULL)
+    {
+        QHash<Compound *, QTreeWidgetItem *>::const_iterator i = Hash.find(compound);
+        if (i != Hash.end() & i.key() == compound) {
+            QTreeWidgetItem* item = i.value();  
+
+                if (item != NULL) {
+                    for (int col = 0; col < treeWidget->columnCount(); col++) {
+                        item->setBackgroundColor(col, QColor(61, 204, 85, 100));
```

Comment:
use https://doc.qt.io/qt-5/qtreewidgetitem.html#setBackground 
setBackgroundColor is [obsolete](https://doc.qt.io/archives/qt-4.8/qtreewidgetitem-obsolete.html)